clean_code/junction.py

```diff
@@ -0,0 +1,81 @@
+from position import Position
+
+
+class Junction():
+    global_refernce = 0
+
+    def __init__(self, position):
+        self.up = None
+        self.down = None
+        self.left = None
+        self.right = None
+        self.position = position
+        self.discovered = False
+        self.finish = False
+        self.start = False
+        self.reference_number = -1
+
+    def get_position(self):
+        return Position(self.position.get_row(), self.position.get_column())
+
+    def get_up(self):
+        return self.up
+
+    def get_down(self):
+        return self.down
+
+    def get_left(self):
+        return self.left
+
+    def get_right(self):
+        return self.right
+
+    def get_reference_number(self):
+        return self.reference_number
+
+    def set_up(self, up):
+        self.up = up
+
+    def set_down(self, down):
+        self.down = down
+
+    def set_left(self, left):
+        self.left = left
+
+    def set_right(self, right):
+        self.right = right
+
+    def set_postion(self, position):
+        self.position = Position(position.get_row(), position.get_column())
+
+    def set_is_discovered(self, is_discovered):
+        Junction.global_refernce += 1
+        self.reference_number = Junction.global_refernce
+        self.discovered = is_discovered
+
+    def set_is_finish(self, is_finish):
```

Comment:
there is no point to call this function with `is_finish=False`. I think you can remove the parameter.